narwhals/_pandas_like/utils.py

```diff
@@ -79,15 +86,65 @@ def validate_dataframe_comparand(index: Any, other: Any) -> Any:
         if other.len() == 1:
             # broadcast
             return other._native_series.iloc[0]
-        if other._native_series.index is not index:
+        if (
+            other._native_series.index is not index
+            and other._implementation is not Implementation.DASK
+        ):
             return set_axis(
                 other._native_series,
                 index,
                 implementation=other._implementation,
                 backend_version=other._backend_version,
             )
         return other._native_series
-    msg = "Please report a bug"  # pragma: no cover
+        return other._series
+    raise AssertionError("Please report a bug")
+
+
+def maybe_evaluate_expr(df: PandasDataFrame, expr: Any) -> Any:
```

Comment:
this, and the function below, are now in `narwhals/_expression_parsing.py`, please remove them from here and use those